src/Build/BackEnd/Components/SdkResolution/SdkResolverManifest.cs

```diff
@@ -21,11 +21,44 @@ internal class SdkResolverManifest
         internal static SdkResolverManifest Load(string filePath)
         {
             using (FileStream fs = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
-            using (XmlDictionaryReader reader = XmlDictionaryReader.CreateTextReader(fs, new XmlDictionaryReaderQuotas()))
             {
-                DataContractSerializer serializer = new DataContractSerializer(typeof(SdkResolverManifest));
-                return (SdkResolverManifest)serializer.ReadObject(reader, true);
+                try
+                {
+                    var doc = new XmlDocument();
+                    doc.Load(fs);
+
+                    if (doc.DocumentElement?.Name == "SdkResolver"
+                        && doc.DocumentElement.NamespaceURI == "")
+                    {
+                        var pathElements = doc.DocumentElement.GetElementsByTagName("Path");
+                        if (pathElements.Count > 0 && pathElements.Item(0) is XmlElement pathElement)
+                        {
+                            return new SdkResolverManifest() { Path = pathElement.InnerText };
+                        }
+                        else
+                        {
+                            ThrowInvalidDocumentNode("Path");
+                        }
+                    }
+                    else
+                    {
+                        ThrowInvalidDocumentNode("SdkResolver");
+                    }
+                }
+                catch(Exception e)
+                {
+                    throw new SerializationException(e.Message, e);
```

Comment:
Won't these be doubly thrown? Thrown from inside `ThrowInvalidDocumentNode`, caught in `catch(Exception e)` here, and re-thrown after wrapping? Could that be simplified, maybe by filtering the `catch` to `where (!(e is SerializationException)`?